src/vos/evt_iter.c

```diff
@@ -566,6 +566,47 @@ int evt_iter_delete(daos_handle_t ih, struct evt_entry *ent)
 
 D_CASSERT(sizeof(((daos_anchor_t *)0)->da_buf) >= sizeof(struct evt_rect));
 
+int
+evt_iter_corrupt(daos_handle_t ih)
+{
+	struct evt_iterator	*iter;
+	struct evt_context	*tcx;
+	struct evt_trace	*trace;
+	struct evt_desc		*desc;
+	int			 rc;
+
+	tcx = evt_hdl2tcx(ih);
+	if (tcx == NULL)
+		return -DER_NO_HDL;
+
+	iter = &tcx->tc_iter;
+	rc = evt_iter_is_ready(iter);
+	if (rc != 0)
+		return rc;
+
+	trace = &tcx->tc_trace[tcx->tc_depth - 1];
+	desc = evt_node_desc_at(tcx, evt_off2node(tcx, trace->tr_node),
+				trace->tr_at);
+	rc = evt_tx_begin(tcx);
+	if (rc != DER_SUCCESS)
+		return rc;
+
+	rc = umem_tx_add(&tcx->tc_umm,
+			 trace->tr_node + offsetof(struct evt_desc, dc_ex_addr),
```

Comment:
why not just 

umem_tx_add_ptr(&tcx->tc_umm, &desc->dc_ex_addr, sizeof(desc->dc_ex_addr)) ?

Or even just the flags field?

Reply:
I don't actually know the umem_tx_add apis all that well and I either copied this or had it recommended to me. In what ways is *_add_ptr better? Is it a cleaner approach?